app/controllers/registrar/invoices_controller.rb

```diff
@@ -61,8 +61,8 @@ def normalize_search_parameters
 
       ca_cache = params[:q][:due_date_lteq]
       begin
-        end_time = params[:q][:due_date_lteq].try(:to_date)
-        params[:q][:due_date_lteq] = end_time.try(:end_of_day)
+        end_time = params[:q][:due_date_lteq]
+        params[:q][:due_date_lteq] = end_time
```

Comment:
I don't understand this without the additonal methods, how is it different than:

```ruby
a = 1
b = a
a = b  # can be skipped
```

The rescue on lines 67-69 also seems weird now, the only error you might see is `NoMethodError` when `q` hash is `Nil`.

Reply:
I postponed refactoring that part of legacy code. There are some other things I don't understand as well.